### PR TITLE
Ar review2

### PR DESCRIPTION
**Pull request by GuillaumeBelz**
Fix comments from the review. WIP

### CODE REVIEW

**Comment by GuillaumeBelz** [comment]
@JamesMBallard @ldanzinger This PR is ready to be reviewed. This contains the fix for the issues described in the Excel doc

File: Common/AR/include/Android/ArCoreWrapper.h [hunk 1]
```diff
@@ -146,7 +146,7 @@ class ArCoreWrapper
   ArCamera* m_arCamera = nullptr;
 
   // data returned from each frame
-  std::array<float, 8> m_transformedUvs;
+  std::array<float, 8> m_transformedUvs = {};
```
Comment (JamesMBallard) [inline Common/AR/include/Android/ArCoreWrapper.h]: Is this needed?

File: Common/AR/source/ArcGISArViewInterface.cpp [hunk 2]
```diff
@@ -73,11 +72,17 @@ ArcGISArViewInterface::ArcGISArViewInterface(bool renderVideoFeed, bool tryUsing
     {
       case Qt::ApplicationSuspended:
         if (m_tracking)
```
Comment (JamesMBallard) [inline Common/AR/source/ArcGISArViewInterface.cpp]: Should `m_tracking` be `m_trackingEnabled`?

`m_tracking` and `m_trackingPaused` seem too similar.

File: Common/AR/source/ArcGISArViewInterface.cpp [hunk 3]
```diff
@@ -378,7 +364,7 @@ void ArcGISArViewInterface::geometryChanged(const QRectF& newGeometry, const QRe
                                                  Qt::InvertedLandscapeOrientation | Qt::InvertedPortraitOrientation);
   }
 
-  QQuickItem::geometryChanged(newGeometry, oldGeometry);
+  QQuickFramebufferObject::geometryChanged(newGeometry, oldGeometry);
```
Comment (JamesMBallard) [inline Common/AR/source/ArcGISArViewInterface.cpp]: Should this just be 
```suggestion
  emit geometryChanged(newGeometry, oldGeometry);
```

File: samples/QmlArExample/qml/main.qml [hunk 4]
```diff
@@ -30,7 +30,11 @@ ApplicationWindow {
     property Camera originCamera: null
 
     // Is true if the current scene is tabletop scene.
-    property bool tabletop: false
+    property bool tabletopMode: false
+
+    // Is true before the initial transformation is not set (tabletop mode) or
+    // before the location data source starts to received data.
+    property bool waitingInitialization: false;
```
Comment (JamesMBallard) [inline samples/QmlArExample/qml/main.qml]:
```suggestion
    property bool waitingForInitialization: false;
```

There may be a better name still. Maybe just `property bool initialized: false;`?

File: Common/AR/include/Android/ArCoreWrapper.h (hunk 1 above)
Comment (GuillaumeBelz) [reply Common/AR/include/Android/ArCoreWrapper.h]: This is a warning from clang-tidy, to default initialize the std::array.

File: Common/AR/source/ArcGISArViewInterface.cpp (hunk 3 above)
Comment (GuillaumeBelz) [reply Common/AR/source/ArcGISArViewInterface.cpp]: This not a signal, this is an overridden function.

File: Plugin/QmlApi/source/QmlArcGISArView.cpp [hunk 5]
```diff
@@ -234,7 +236,7 @@ void QmlArcGISArView::setHeadingInternal(double heading)
  */
 void QmlArcGISArView::resetTrackingInternal()
 {
-  // to implmente
+  // to implemente
```
Comment (JamesMBallard) [inline Plugin/QmlApi/source/QmlArcGISArView.cpp]:
```suggestion
  // to implement
```

**Comment by GuillaumeBelz** [comment]
I will merge this PR and create a new PR if there are more changes to do.

**Pull request merged**